kedro/io/core.py

```diff
@@ -290,11 +351,33 @@ def save(self: Self, data: _DI) -> None:
         return save
 
     def __init_subclass__(cls, **kwargs: Any) -> None:
-        """Decorate the `load` and `save` methods provided by the class.
+        """Customizes the behavior of subclasses of AbstractDataset during
+        their creation. This method is automatically invoked when a subclass
+        of AbstractDataset is defined.
 
+        Decorates the `load` and `save` methods provided by the class.
         If `_load` or `_save` are defined, alias them as a prerequisite.
 
         """
+
+        # Save the original __init__ method of the subclass
+        init_func: Callable = cls.__init__
+
+        def new_init(self, *args, **kwargs) -> None:  # type: ignore[no-untyped-def]
```

Comment:
```suggestion
        @wraps(init_func)
        def new_init(self, *args, **kwargs) -> None:  # type: ignore[no-untyped-def]
```

I think we need this in order to make sure that `cls.__init__.__name__` and `cls.__init__.__doc__` are correct even if we wrap it with `new_init`. For more information, see: https://docs.python.org/3/library/functools.html#functools.wraps